src/gleam/string.gleam

```diff
@@ -485,13 +480,24 @@ fn do_join(strings: List(String), separator: String) -> String {
 /// // -> "121"
 /// ```
 ///
-pub fn pad_left(string: String, to desired_length: Int, with pad_string: String) {
+pub fn pad_left(
+  string: String,
+  to desired_length: Int,
+  with pad_string: String,
+) -> String {
   let current_length = length(string)
   let to_pad_length = desired_length - current_length
-  padding(to_pad_length, pad_string)
-  |> iterator.append(iterator.single(string))
-  |> iterator.to_list
-  |> concat
+
+  case to_pad_length <= 0 {
+    True -> string
+    False ->
+      case length(pad_string) {
+        1 -> repeat(pad_string, to_pad_length)
+        pad_string_length ->
+          padding(to_pad_length, pad_string, pad_string_length)
+      }
+      <> string
+  }
```

Comment:
I'd write this as 
```gleam
repeat(to_pad_length, pad_string) <> string
```
And change the `padding` function itself to:
```gleam
fn padding(size: Int, pad_string: String) -> String {
  let num_pads = size / pad_string_length
  let extra = size % pad_string_length
  repeat(pad_string, num_pads) <> slice(pad_string, 0, extra)
}
```
`slice` should already be a no-op if the length is zero, otherwise that's a bug in `slice` and should be fixed there instead of special casing it inside the `padding` function.

Reply:
I've dropped the special casing for a single-character as its not needed now that string slicing is optimised.